File: 4. Binary Tree/2_breadth_first_values.java
import java.util.List;
import java.util.ArrayList;
import java.util.LinkedList;
import java.util.Queue;

class Node<T> {
    T val;
    Node<T> left;
    Node<T> right;

    public Node(T val) {
        this.val = val;
        this.left = null;
        this.right = null;
    }
}

class Source {
    public static List<String> breadthFirstValues(Node<String> root) {
        List<String> result = new ArrayList<>();
        Queue<Node<String>> queue = new LinkedList<>();
        Node<String> current = root;

        if (root != null) {
            queue.add(current);
        }

        while (!queue.isEmpty()) {
            current = queue.remove(); // first in -> first out
            result.add(current.val);

            if (current.left != null) {
                queue.add(current.left);
            }

            if (current.right != null) {
                queue.add(current.right);
            }
        }

        return result;
    }
}

File: 4. Binary Tree/2_breadth_first_values.py
from collections import deque

class Node:
    def __init__(self, val):
        self.val = val
        self.left = None
        self.right = None

def breadthFirstValues(root):
    result = []
    if root is None:
        return result

    queue = deque([root])
    while queue:
        current = queue.popleft()
        result.append(current.val)
        if current.left:
            queue.append(current.left)
        if current.right:
            queue.append(current.right)

    return result
